Support Termux in ansible.builtin.package

cilki opened the pull request:
##### SUMMARY

This enables usage of `ansible.builtin.package` instead of `ansible.builtin.apt` in Termux environments (Android). Tested on Pixel Fold and Pixel 6.

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

Example playbook:
```
- hosts: example
  tasks:
    - name: Install packages
      ansible.builtin.package:
        update_cache: yes
        name:
          - gitui
```

Fails because it can't find `apt`:

```paste below
TASK [Install packages] 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoneType: None
fatal: [example]: FAILED! => {"changed": false, "msg": "Could not find a module for unknown."}
```

Fixed after adding the right path:

```
TASK [Install packages] 
changed: [example]
```

s-hertel commented on lib/ansible/module_utils/facts/system/pkg_mgr.py:
```diff
@@ -47,6 +47,7 @@
             {'path': '/usr/sbin/sorcery', 'name': 'sorcery'},
             {'path': '/usr/bin/installp', 'name': 'installp'},
             {'path': '/QOpenSys/pkgs/bin/yum', 'name': 'yum'},
+            {'path': '/data/data/com.termux/files/usr/bin/apt', 'name': 'apt'},
```
Do you know if this path is reliable, or depends on the configuration of the Android device?

sivel replied:
I know you can move, or explicitly install apps onto other disks, such as an extra SD card. So would that make the path different, or is this some how a chrooted path inside the app that always exists?

cilki replied:
According to their wiki, https://github.com/termux/termux-packages/wiki/Termux-file-system-layout, it seems to be a stable path. Termux exports a `${PREFIX}` variable, but it's not configurable according to [this](https://wiki.termux.com/wiki/Differences_from_Linux):

> The root file system and user home directory are located in a private application data directory which lives on the /data partition. Paths to these directories are exposed as $PREFIX and $HOME respectively.

> You cannot move $PREFIX to another location because all programs expect that $PREFIX will not be changed. Additionally, you cannot have binaries, symlinks and other files from $PREFIX on sdcard because that file system does not support unix permissions, symlinks, sockets, etc... 

As an aside, thanks for all of the work on Ansible. This tool has changed my life :+1: !

s-hertel reviewed (changes requested):
This will need a changelog.

cilki commented:
> This will need a changelog.

Let me know if I did that changelog wrong. Thanks

cilki committed:
Support Termux in ansible.builtin.package

This enables usage of `ansible.builtin.package` instead of `ansible.builtin.apt`
in Termux environments (Android). Tested on Pixel Fold and Pixel 6.
